### PR TITLE
fix: result type generation

### PR DESCRIPTION
**Pull request by Zercerium**
I think this could be a potential fix for https://github.com/tauri-apps/tauri-bindgen/issues/143

### CODE REVIEW

File: crates/gen-js/src/lib.rs [hunk 1]
```diff
@@ -83,10 +83,10 @@ pub trait JavaScriptGenerator {
             Type::Result { ok, err } => {
                 let ok = ok
                     .as_ref()
-                    .map_or("() => {}".to_string(), |ty| self.print_deserialize_ty(ty));
+                    .map_or("() => {}".to_string(), |ty| "(de) => ".to_string() + &self.print_deserialize_ty(ty));
```
Comment (JonasKruckenberg) [inline crates/gen-js/src/lib.rs:86]:
```suggestion
                    .map_or("() => {}".to_string(), |ty| format!("(de) => {}", self.print_deserialize_ty(ty)));
```

I prefer `format!` for clarity

**Review by JonasKruckenberg — APPROVED** [review]

File: crates/gen-js/src/lib.rs [hunk 2]
```diff
@@ -83,10 +83,10 @@ pub trait JavaScriptGenerator {
             Type::Result { ok, err } => {
                 let ok = ok
                     .as_ref()
-                    .map_or("() => {}".to_string(), |ty| self.print_deserialize_ty(ty));
+                    .map_or("() => {}".to_string(), |ty| format!("(de) => {}", self.print_deserialize_ty(ty)));
                 let err = err
                     .as_ref()
-                    .map_or("() => {}".to_string(), |ty| self.print_deserialize_ty(ty));
+                    .map_or("() => {}".to_string(), |ty| "(de) => ".to_string() + &self.print_deserialize_ty(ty));
```
Comment (JonasKruckenberg) [inline crates/gen-js/src/lib.rs:89]: Same here btw 😉

**Review by JonasKruckenberg — CHANGES_REQUESTED** [review]

File: crates/gen-js/src/lib.rs (hunk 2 above)
Comment (Zercerium) [reply crates/gen-js/src/lib.rs]: ahh, sry

**Review by JonasKruckenberg — APPROVED** [review]

**Comment by JonasKruckenberg** [comment]
Thank you!

**Pull request merged**